p2p/conn/secret_connection.go

```diff
@@ -237,6 +243,52 @@ func shareEphPubKey(conn io.ReadWriteCloser, locEphPub *[32]byte) (remEphPub *[3
 	return &_remEphPub, nil
 }
 
+// use the samne blacklist as lib sodium (see https://eprint.iacr.org/2017/806.pdf for reference):
+// https://github.com/jedisct1/libsodium/blob/536ed00d2c5e0c65ac01e29141d69a30455f2038/src/libsodium/crypto_scalarmult/curve25519/ref10/x25519_ref10.c#L11-L17
+var blacklist = [][32]byte{
```

Comment:
I'll check the paper later today, but this list seems incomplete. IIRC, the multiplicative group of Curve25519 is of order `8p`, so there should be 8 points in the subgroup `<g^p>`, whereas only 3 or 4 points from the subgroup are listed here. (The last 3 points are just non-canonical forms, didn't check the 4th to last)

Instead of blacklisting like this, may just be easier to check if a point is low order. (`x^8 == 0`)

Reply:
You are right that there are more points which aren't valid. This method is just a band-aid that detects some funky behaviour. According to djb even these exclusions are not necessary for DH: https://cr.yp.to/ecdh.html#validate
But there are other takes: https://research.kudelskisecurity.com/2017/04/25/should-ecdh-keys-be-validated/

In any case, AFAIU, we should additionally authenticate the transcript as Tony suggested in the discussion of https://github.com/tendermint/tendermint/issues/3010#issuecomment-447043829. This probably makes worrying about invalid / low-order points in this context obsolete.